Adding details on user locator behavior (#5952)
DataDog/documentation: content/en/synthetics/browser_tests.md

| `Test the content of the URL of the active page`          | Takes the URL of the last page that was interacted with, then asserts whether a specific value (`string`, `number`, `regex`) is present within it.                                     |
| `Test that an email was received`                         | Asserts that an email was sent and whether specific values (`string`, `number`, `regex`) are present within the email subject or body. This assertion leverages  [email variables][9]. |

> Advanced options for assertions allow you to specify an X-path or a CSS class or ID that you want to use to perform the element selection for any HTML element. For example, `div`, `h1`, or `.hero-body`. Once you define an element, hit **Test**, and it will highlight the element in the recording on the right.  

Reviewer: ```suggestion
Advanced options for assertions allow you to specify an X-path or a CSS class or ID that you want to use to perform the element selection for any HTML element. For example, `div`, `h1`, or `.hero-body`. Once you define an element, hit **Test** to highlight the element in the recording on the right.  
```

> If the defined user locator fails, by default the test will be considered as failed. You can decide to fallback on the regular Browser tests algorithm by unticking the `If user specified locator fails, fail test` box.

Reviewer: ```suggestion
If the defined user locator fails, by default the test is considered as failed. You can decide to fallback on the regular Browser tests algorithm by unticking the `If user specified locator fails, fail test` box.
```
